### PR TITLE
Use .editorconfig

### PR DESCRIPTION
**Pull request by Zonnex**
Formatted code with initial config mentioned in #99

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+# This file is the top-most EditorConfig file
+root = true
+
+# All Files
+[*]
+charset = utf-8
+end_of_line = crlf
+indent_style = space
+indent_size = 4
+insert_final_newline = false
```
Comment (fredrik-lundin) [inline .editorconfig]: `false`? really..? really..!?

Comment (viktorvan) [reply .editorconfig]: Agree, I would prefer **insert_final_newline = true** and **end_of_line = lf**

Comment (Zonnex) [reply .editorconfig]: Why use one type of setting over another in this case?

Comment (Zonnex) [reply .editorconfig]: https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-file

These are the reasons?

**Comment by Zonnex** [comment]
I've used [Resharper CodeCleanup CLI](https://blog.jetbrains.com/dotnet/2018/03/01/code-cleanup-resharper-command-line-tools/) to try apply rules to all files (since dotnet format only applies whitespace formatting).

One thing it does however is applying some spacing rules that that turns this. 
```new AuthenticationProperties { RedirectUri = "/" }```
into this
```new AuthenticationProperties {RedirectUri = "/"}```

I haven't found where the setting is yet, but I'll find it soon.
One more thing it does is reorder private fields, I'll see if I can disable that. I'd like to have it just run all rules defined in editorconfig but no success yet.

File: samples/IdentityServer.ClientSample/Startup.cs [hunk 2]
```diff
@@ -66,11 +67,8 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
         {
             app.UseHttpsRedirection();
 
-            if (env.IsDevelopment())
-            {
-                app.UseDeveloperExceptionPage();
-            }
-            
+            if (env.IsDevelopment()) app.UseDeveloperExceptionPage();
```
Comment (PeterOrneholm) [inline samples/IdentityServer.ClientSample/Startup.cs]: I usually don't like these on one line, what do you think?

File: samples/IdentityServer.ServerSample/Controllers/AccountController.cs [hunk 3]
```diff
@@ -23,8 +24,8 @@ public class AccountController : Controller
 
         public IActionResult Login(string returnUrl)
         {
-            var schemes = _authenticationSchemeProvider.GetAllSchemesAsync();
-            var providers = schemes.Result
+            Task<IEnumerable<AuthenticationScheme>> schemes = _authenticationSchemeProvider.GetAllSchemesAsync();
```
Comment (PeterOrneholm) [inline samples/IdentityServer.ServerSample/Controllers/AccountController.cs]: Hmm, i had missed this, but why are we not doing async/await?

File: samples/IdentityServer.ClientSample/Startup.cs (hunk 2 above)
Comment (viktorvan) [reply samples/IdentityServer.ClientSample/Startup.cs]: If it’s less than 120 chars wide I am fine with a one liner, but I still like to use braces.

Comment (Zonnex) [reply samples/IdentityServer.ClientSample/Startup.cs]: Hi, thanks this one is fixed now.

File: samples/IdentityServer.ServerSample/Controllers/AccountController.cs (hunk 3 above)
Comment (Zonnex) [reply samples/IdentityServer.ServerSample/Controllers/AccountController.cs]: No idea actually! It should be

**Comment by Zonnex** [comment]
Regarding var vs types. I currently try to follow the [corefx guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md). Thoughts?

File: .editorconfig (hunk 1 above)
Comment (fredrik-lundin) [reply .editorconfig]: Basically yes. But mostly it's just a cosmetic thing for me. I'm so used to seeing new lines at the end of a file, so when it's not there, it looks messy and compact.

Note that the editor config for roslyn (https://github.com/dotnet/roslyn/blob/master/.editorconfig) that you linked to in the related issue sets `insert_final_newline = true`. It's a very common thing to do.

File: samples/IdentityServer.ClientSample/Startup.cs (hunk 2 above)
Comment (Zonnex) [reply samples/IdentityServer.ClientSample/Startup.cs]: My bad, I missed the brace preference. I usually prefer single line statements/expressions to be without braces but we can put in preference for.

**Comment by fredrik-lundin** [comment]
> I've used [Resharper CodeCleanup CLI](https://blog.jetbrains.com/dotnet/2018/03/01/code-cleanup-resharper-command-line-tools/) to try apply rules to all files (since dotnet format only applies whitespace formatting).
> 
> One thing it does however is applying some spacing rules that that turns this.
> `new AuthenticationProperties { RedirectUri = "/" }`
> into this
> `new AuthenticationProperties {RedirectUri = "/"}`
> 
> I haven't found where the setting is yet, but I'll find it soon.
> One more thing it does is reorder private fields, I'll see if I can disable that. I'd like to have it just run all rules defined in editorconfig but no success yet.

Have you tried playing around with `spaces_around_brackets`

Looks like it could be what you are looking for (https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties):
image

File: samples/IdentityServer.ServerSample/Startup.cs [hunk 4]
```diff
@@ -108,47 +102,47 @@ public void ConfigureServices(IServiceCollection services)
...
+                        {
+                            options.IssueBirthdateClaim = true;
+                            options.IssueGenderClaim = true;
+                        })
+                        .AddBankIdSameDevice(GrandIdAuthenticationDefaults.BankIdSameDeviceAuthenticationScheme,
+                            "GrandID (SameDevice)", options => { })
+                        .AddBankIdOtherDevice(GrandIdAuthenticationDefaults.BankIdOtherDeviceAuthenticationScheme,
+                            "GrandID (OtherDevice)", options => { })
+                        .AddBankIdChooseDevice(GrandIdAuthenticationDefaults.BankIdChooseDeviceAuthenticationScheme,
+                            "GrandID (ChooseDevice)", options => { });
```
Comment (Zonnex) [inline samples/IdentityServer.ServerSample/Startup.cs]: Bad wrapping here

File: samples/IdentityServer.ServerSample/Startup.cs [hunk 5]
```diff
@@ -108,47 +102,47 @@ public void ConfigureServices(IServiceCollection services)
...
 
                     if (Configuration.GetValue("ActiveLogin:GrandId:UseSimulatedEnvironment", false))
-                    {
                         builder.UseSimulatedEnvironment();
-                    }
                     else
-                    {
                         builder.UseProductionEnvironment(config =>
```
Comment (Zonnex) [inline samples/IdentityServer.ServerSample/Startup.cs]: Lack of braces here looks bad I think

File: samples/IdentityServer.ClientSample/Startup.cs (hunk 2 above)
Comment (Zonnex) [reply samples/IdentityServer.ClientSample/Startup.cs]: I have changed the editorconfig to have ```csharp_prefer_braces = true:suggestion```

**Comment by fredrik-lundin** [comment]
Any reasons this PR never got completed? @PeterOrneholm @Zonnex

**Comment by Zonnex** [comment]
@nikolaykrondev 
`We only use var when it's obvious what the variable type is (e.g. var stream = new FileStream(...) not var stream = OpenStandardInput()).`
The idea is that code reviewing outside an editor is slightly annoying if you can't easily see what the types are. Types makes the code easier to reason about.

**Comment by Zonnex** [comment]
We recently did this in a few projects, and my suggestion is to generate a new editorconfig, recommit the changes and we'll review this again after

**Comment by viktorvan** [comment]
Well nowadays you can just run gitpod.io on your repo, and get an IDE in the browser, so that is almost getting to be a moot point. 

https://gitpod.io/#https://github.com/ActiveLogin/ActiveLogin.Authentication

`var` isn’t just a shorter syntax, it does come with some other benefits.

**Comment by PeterOrneholm** [comment]
This is so out of date with master, so I'll close it, and we can make a new one when implementing it.

**Pull request closed without merging**